python-frontend/src/main/java/org/sonar/python/tree/CallExpressionImpl.java

```diff
@@ -203,8 +209,15 @@ public PythonType typeV2() {
           result = UnionType.or(result, functionType.returnType());
         }
       }
-      return result == null ? PythonType.UNKNOWN : new ObjectType(result);
+      return result == null ? PythonType.UNKNOWN : new ObjectType(result, typeSource);
     }
     return PythonType.UNKNOWN;
   }
+
+  TypeSource computeTypeSource() {
+    if (callee instanceof QualifiedExpression qualifiedExpression) {
+      return qualifiedExpression.qualifier().typeV2().typeSource();
+    }
```

Comment:
I guess in some cases we can compute the type source out of `Name` like in the example below:
```python
class A:
    def __call__(self) -> int:
        return 42
    
def foo(a: A):
    a.__call__() # the type source is coming from qualified expression and it is TYPE_HINT
    a() # the type source is coming from name and it is the same - TYPE_HINT
```